base-deploy

```diff
@@ -87,7 +87,7 @@ for token in $tokens; do
         . "load-ilk-$(echo "$token" | tr '[:upper:]' '[:lower:]')-$(echo "$ilk" | tr '[:upper:]' '[:lower:]')-$(seth chain)"
         rm "load-ilk-$(echo "$token" | tr '[:upper:]' '[:lower:]')-$(echo "$ilk" | tr '[:upper:]' '[:lower:]')-$(seth chain)"
 
-        if [ "${token}" != "ETH" ] && [ $newToken = "true" ]; then
+        if [ "$token" != "ETH" ] && [ $newToken == "true" ]; then
```

Comment:
Note that only variables need to be wrapped in double quotes here as these are arguments passed to the `[[` command. Sort of the exact opposite of a programmers intuition :P. If a variable is not quoted and empty it would be rendered as e.g. `[[ == ETH ]]` and therefor syntactically incorrect according to `[[`.

```suggestion
        if [[ "$token" != ETH && "$newToken" == true ]]; then
```

Reply:
Ahh ok, so we need to still quote the variables.

Reply:
@icetan is `true` a special case though? Since it's also a command.

Reply:
Not really, think of it like invoking any program in bash. Where everything except the first word is just a literal string passed as an argument to the command (first word).

It might be clearer to use the `test` command which is synonymous with `[` as an example:

This will echo `hello` even though it looks like the test will fail, because `test` returns `0` if it gets any argument here and `false` is not evaluated as the command `false` but just as any argument to any program.

```
test false && echo hello
```

The `test` command will get the args `$1` to equal `true`, `$2` to equal `==` and `$3` to equal `false` in this case:

```
test                   true == false
^- executable         ^-------------^- argument list
```

Reply:
@asymmetric @gbalabasquer 

So I learnt some more about this today, with `[` you have to quote like my statement above but with extended test (i.e. `[[`) you don't :P

Basically `[` is a POSIX compatible command which should be used for portability and `[[` is the extended version. As long as we use something like Nix to make sure you are running the scripts with a specific shell we can use `[[`. But if we can't guarantee the environment shell we should probably keep to POSIX standard `[`.

Very thorough description: https://stackoverflow.com/a/31366734

Reply:
Right, that was what I had read and why I had moved initially everything to `[`. Although with the mac bash `3.2.57` `[[` works fine.

Reply:
Isn't `[[` a Bash extension? Don't think we support other shells, so for all intents and purporses `[[` is portable for us as well?

Reply:
Yes, I'm just saying that without a standard to follow we need to know what the lowest common denominator is. Is it bash 3?

Even subtle differences in behavior or features could render hard to debug issues. A special when writing non-type safe scripts like bash. You wouldn't even know there is an issues until that specific path of the script is taken in a specific environment.